### ISSUE TITLE
when using screen-print in query.iterator may loss message

### ISSUE THREAD

**Issue by kyrosle** [nicopap/bevy-debug-text-overlay#2]
hello , i feel confused when i am using the `screen_paint!` in my query in the system function, my query type is such as `query: Query<&GlobalTrasnform>`, then the using postion is `screen_print!("{:?}", trans.translation())`

**Comment by nicopap** [comment]
Yeah, each `screen_print!` call has a "slot." And calling it repetitively will overwrite the previous value, even within the same frame. The workaround is to first collect your iterator and then `screen_print!` the resulting value.

```rust
// WORKAROUND:
// do not do this
for foo in &iterator {
  screen_print!("{foo:?}");
}
// do this instead
let foos: Vec<_> = iterator.iter().collect();
screen_print!("{foos:#?}");
```

This is not a trivial fix because, I need to be able to "identify" the line to print, so that calling it once per frame doesn't spawn an additional line per frame, but rather just replaces the previous line.

**Comment by kyrosle** [comment]
all right, thanks for you reply

**Comment by thefakeplace** [comment]
Would you consider changing this? I'm using this for my game and I would like to log messages when clients connect.

Naturally, if two clients connect in quick succession, I don't see what I expect. Maybe an optional parameter?

**Comment by nicopap** [comment]
I've missed this functionality myself a bunch. This is a good reason to add it.

**Comment by nicopap** [comment]
Fixed by 0e19579

**Comment by thefakeplace** [comment]
Wow, awesome! Thank you so much!
